src/stores/messages/MessagesDataProviderStore.ts

```diff
@@ -345,6 +339,7 @@ export default class MessagesDataProviderStore implements MessagesDataStore {
 
 		const nextMessages = messages
 			.slice(0, messages.length - prevMessages.length)
+			.filter(msg => !this.messages.map(msg => msg.messageId).includes(msg.messageId))
```

Comment:
i thought onNextChannelResponse can be simplified a lot since we dont have to worry about order now
it should just insert new messages in the list without any extra checks

would it work like so?
```
public onNextChannelResponse = (nextMessages) => {
      this.lastNextChannelResponseTimestamp = null;
       this.messages = [...nextMessages, ...this.messages]
}

```